galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+    @param interpolant      Either an Interpolant instance or a string indicating which interpolant
+                            should be used.  Options are 'nearest', 'sinc', 'linear', 'cubic',
+                            'quintic', or 'lanczosN' where N should be the integer order to use.
+                            [default: galsim.Quintic()]
+    @param pad_factor       Additional multiple by which to zero-pad the PSF image to avoid folding
+                            compared to what would be employed for a simple Airy.  Note that
+                            `pad_factor` may need to be increased for stronger aberrations, i.e.
+                            those larger than order unity.  [default: 1.5]
+    @param oversampling     Optional oversampling factor for the wrapped InterpolatedImage. Setting
+                            `oversampling < 1` will produce aliasing in the PSF (not good).
+                            Usually `oversampling` should be somewhat larger than 1.  1.5 is
+                            usually a safe choice.  [default: 1.5]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+    """
+    def __init__(self, screen_list, lam=500., exptime=0.0, flux=1.0,
+                 theta_x=0.0*galsim.arcmin, theta_y=0.0*galsim.arcmin,
+                 scale_unit=galsim.arcsec, interpolant=None,
+                 diam=8.4, obscuration=0.6,
+                 pad_factor=1.5, oversampling=1.5,
+                 _pupil_size=None, _pupil_scale=None,
+                 gsparams=None, _eval_now=True, _bar=None):
+
+        self.screen_list = screen_list
+        self.lam = lam
+        self.exptime = exptime
+        self.theta_x = theta_x
+        self.theta_y = theta_y
+        self.scale_unit = scale_unit
+        self.interpolant = interpolant
+        self.diam = diam
+        self.obscuration = obscuration
+        self.pad_factor = pad_factor
+        self.oversampling = oversampling
+
+        if _pupil_scale is None:
+            # Generically, Galsim propagates stepK() for convolutions using
+            #   scale = sum(s**-2 for s in scales)**(-0.5)
+            # We're not actually doing convolution between screens here, though.  In fact, the right
+            # relation for Kolmogorov screens uses exponents -5./3 and -3./5:
+            #   scale = sum(s**-5./3 for s in scales)**(-3./5)
+            # Since most of the layers in a PhaseScreenList are likely to be (nearly) Kolmogorov
+            # screens, we'll use that relation.
+            _pupil_scale = (sum(layer.pupil_scale(lam)**(-5./3) for layer in screen_list))**(-3./5)
+            _pupil_scale /= oversampling
+        self._pupil_scale = _pupil_scale
+        if _pupil_size is None:
+            _pupil_size = self.diam * self.pad_factor
+        self._nu = int(np.ceil(_pupil_size/self._pupil_scale))
+        self._pupil_size = self._nu * self._pupil_scale
```

Comment:
I found this bit confusing.  It could probably benefit from an in-code comment.  Specifically that pupil_size is not the size of the pupil, but rather the physical size of the region that will be covered by the numpy array, aper.

Also, nu sounds like a non-integer constant.  Recommend switching to npix or similar to be clearer that this is the number of pixels in each direction in the aperture array.